inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/detection_output_gpu_ref.cl

```diff
@@ -0,0 +1,729 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "include/data_types.cl"
+#include "include/fetch_data.cl"
+#include "include/common.cl"
+#include "include/detection_output_common.cl"
+
+#define unroll_for __attribute__((opencl_unroll_hint)) for
+#define NUM_CLASSES_ACC (NUM_CLASSES + 1)
+
+typedef struct {
+    int batchId;
+    int classId;
+    int boxId;
+    UNIT_TYPE score;
+} FUNC(Scores);
+
+#define SCORES_INFO FUNC(Scores)
+
+inline void FUNC(swap_scores_info)(__global SCORES_INFO* a, __global SCORES_INFO* b)
+{
+    SCORES_INFO temp = *a;
+    *a = *b;
+    *b = temp;
+}
+
+inline int FUNC(partition)(__global SCORES_INFO* arr, int l, int h, bool use_custom_comp)
+{
+    UNIT_TYPE pivotScore = arr[h].score;
+    int pivotBoxId = arr[h].boxId;
+    int i = (l - 1);
+    for (int j = l; j <= h - 1; j++) {
+        if (use_custom_comp) {
+            if ((arr[j].score > pivotScore) || (arr[j].score == pivotScore && arr[j].boxId < pivotBoxId)) {
+                i++;
+                FUNC_CALL(swap_scores_info)(&arr[i], &arr[j]);
+            }
+        } else {
+            if (arr[j].score > pivotScore) {
+                i++;
+                FUNC_CALL(swap_scores_info)(&arr[i], &arr[j]);
+            }
+        }
+    }
+    FUNC_CALL(swap_scores_info)(&arr[i + 1], &arr[h]);
+    return (i + 1);
+}
+
+inline void FUNC(bubbleSortIterative)(__global SCORES_INFO* arr, int l, int h)
+{
+    for (int i = 0; i < h-l; i++) {
+        bool swapped = false;
+        for (int j = l; j < h-i; j++) {
+            if ((arr[j].score > arr[j+1].score) || (arr[j].score == arr[j+1].score && arr[j].boxId < arr[j+1].boxId)) {
+                FUNC_CALL(swap_scores_info)(&arr[j], &arr[j+1]);
+                swapped = true;
+            }
+        }
+
+        if (!swapped)
+            break;
+    }
+}
+
+inline void FUNC(quickSortIterative)(__global SCORES_INFO* arr, int l, int h, bool use_custom_comp)
+{
+    // Create an auxiliary stack
+    const int kStackSize = 100;
```

Comment:
how about use uchar as the dtype of stack and increase stack size to 200?

Reply:
In some model, data of the stack exceeds uchar's data range. (e.g. l=249, h=299 in faster-rcnn-zf). and in general case, when executing quickSortIterative on each work items after partitioning, the size of stack does not exceed.(>100)

Reply:
If it exeeds the stack size, it will use bubblesort isnt it?

Reply:
Yes, I know that bubble sort was just added to handle unexpected exception case(top exceeds bound of stack size)

Reply:
so if the top > stack_size than it will not use stack, so there will be no problem?

Reply:
Ah I was confused with something others. The value should be able to represent all the numbers.